bravado/mapping/exception.py

```diff
@@ -0,0 +1,9 @@
+class SwaggerError(Exception):
+    """Raised when an error is encountered."""
+
+    def __init__(self, msg, cause=None):
+        """
+        :param msg: String message for the error.
+        :param cause: Optional exception that caused this one.
+        """
+        super(Exception, self).__init__(msg, cause)
```

Comment:
I'm trying to reduce the coupling between `bravado` and `bravado.mapping` in hopes of making `bravado.mapping` easier to use in pyramid_swagger.  Currently all references to `SwaggerError` are in `bravado.mapping`.
```
spatel@vm-yelp:~/git/swagger-py-analogue (cleanup) $ git grep SwaggerError
bravado/mapping/docstring.py:from bravado.mapping.exception import SwaggerError
bravado/mapping/docstring.py:        raise SwaggerError("No proper type could be found from %s" % spec)
bravado/mapping/exception.py:class SwaggerError(Exception):
bravado/mapping/marshal.py:from bravado.mapping.exception import SwaggerError
bravado/mapping/marshal.py:        raise SwaggerError('Unknown type {0} for value {1}'.format(
bravado/mapping/operation.py:from bravado.mapping.exception import SwaggerError
bravado/mapping/operation.py:    :raises: SwaggerError when the status_code could not be mapped to a response
bravado/mapping/operation.py:        raise SwaggerError(
bravado/mapping/param.py:from bravado.mapping.exception import SwaggerError
bravado/mapping/param.py:        raise SwaggerError(
bravado/mapping/schema.py:from bravado.mapping.exception import SwaggerError
bravado/mapping/schema.py:                    raise SwaggerError(
bravado/mapping/unmarshal.py:from bravado.mapping.exception import SwaggerError
bravado/mapping/unmarshal.py:        raise SwaggerError(
tests/mapping/docstring/formatted_type_test.py:from bravado.mapping.exception import SwaggerError
tests/mapping/docstring/formatted_type_test.py:    with pytest.raises(SwaggerError) as excinfo:
tests/mapping/marshal/validate_primitive_test.py:# from bravado.exception import SwaggerError
tests/mapping/marshal/validate_primitive_test.py:#     with pytest.raises(SwaggerError) as excinfo:
tests/mapping/operation/get_response_spec_test.py:from bravado.mapping.exception import SwaggerError
tests/mapping/operation/get_response_spec_test.py:    with pytest.raises(SwaggerError) as excinfo:
tests/resource_response_test.py:from bravado.mapping.exception import SwaggerError
tests/resource_response_test.py:        self.assertRaises(SwaggerError, SwaggerClient.from_url,
```

I'd like to revisit specific exceptions at the `bravado` scope (connection but not mapping realted) eventually.